packages/libs/lambda-at-edge/src/regeneration-handler.ts

```diff
@@ -0,0 +1,86 @@
+import lambdaAtEdgeCompat from "@sls-next/next-aws-cloudfront";
+import { OriginRequestDefaultHandlerManifest } from "./types";
+import { S3Client } from "@aws-sdk/client-s3";
+import { buildS3RetryStrategy } from "./s3/s3RetryStrategy";
+
+export const handler: AWSLambda.SQSHandler = async (event) => {
+  await Promise.all(
+    event.Records.map(async (record) => {
+      const bucketName = record.messageAttributes.BucketName?.stringValue;
+      const bucketRegion = record.messageAttributes.BucketRegion?.stringValue;
+      const manifestString = record.messageAttributes.Manifest?.stringValue;
+      const basePath = record.messageAttributes.BasePath?.stringValue;
+      const cloudFrontEventRequestString =
+        record.messageAttributes.CloudFrontEventRequest?.stringValue;
+      if (
+        !bucketName ||
+        !bucketRegion ||
+        !cloudFrontEventRequestString ||
+        !manifestString
+      ) {
+        throw new Error(
+          "Expected BucketName, BucketRegion, CloudFrontEventRequest & EnableHTTPCompression message attributes"
+        );
+      }
+      const cloudFrontEventRequest: AWSLambda.CloudFrontRequest = JSON.parse(
+        cloudFrontEventRequestString
+      );
+      const manifest: OriginRequestDefaultHandlerManifest = JSON.parse(
+        manifestString
+      );
+
+      const s3 = new S3Client({
+        region: cloudFrontEventRequest.origin?.s3?.region,
+        maxAttempts: 3,
+        retryStrategy: await buildS3RetryStrategy()
+      });
+
+      const { req, res } = lambdaAtEdgeCompat(
```

Comment:
This looks to be just regular Lambda, so this should not be needed, right? You can just create a Node req, res in that case?

Reply:
Correct! Good catch.